### ISSUE TITLE
DIV Strange page-break if doesn't fit on a page.

### ISSUE THREAD

**Issue by Nagrom33** [dompdf/dompdf#2570]
I have a problem if my `<div>` doesn't fit, than i got an strange page break.
I think it's something with the padding. Can someone help me out?

HTML:
```
<div class="info">
      <h2 class="font--xxl margin--m">Creatable checkboxes tooltip</h2>
      <p class="font--m margin--m">Lorem creatable checkbox lorem</p>
</div>
```

CSS:
```
.info {
  width: 570px;

  padding: 32px 48px 40px;
  border-radius: 8px;
  margin-top: 24px;

  color: #000;
  background: #d9ebf7;

  page-break-inside: avoid;
}
```

Example of export:
<a href="https://ibb.co/ZdycfsV"><img src="https://i.ibb.co/TczPwXk/Schermafbeelding-2021-09-23-om-14-15-13.png" alt="Schermafbeelding-2021-09-23-om-14-15-13" border="0"></a>

**Comment by bsweeney** [comment]
FYI easier to debug with a full example, this seems to do it:

```
<style>
    @page { size: 650px 185px; margin: 0px; }
    .info {
      width: 570px;
    
      padding: 32px 48px 40px;
      border-radius: 8px;
      margin-top: 24px;
    
      color: #000;
      background: #d9ebf7;
    
      page-break-inside: avoid;
    }
</style>

<div class="info">
        <h2 class="font--xxl margin--m">Creatable checkboxes tooltip</h2>
        <p class="font--m margin--m">Lorem creatable checkbox lorem</p>
</div>
```

There does seem to be an issue here since the containing element is pushed to the next page but still has a presence on the previous page. There also seems to be an issue determining whether or not it's appropriate to push to the next page (as in my example where it's the only element in the document).

The upcoming 1.1.0 is much improved in this regard but still appears to produce a suboptimal rendering.

**Comment by Mellthas** [comment]
`page-break-inside` is only checked if the frame has a previous sibling. This should be addressed by #2571.

**Comment by Nagrom33** [comment]
Thanks @Mellthas this works:

```
.info {
  page-break-inside: avoid !important;
}

.info > *:first-child {
  page-break-before: avoid !important;
}

.info > *:last-child {
    page-break-after: avoid !important;
}
```
